io/net/net_tools.py

```diff
@@ -0,0 +1,374 @@
+            raise AssertionError("%s package is need to test" % pkg)
+
+
+class Hostname(Test):
+    """
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    Test to verify the functionality of the hostname command
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.restore_hostname = False
+        # Get Hostname
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            # set hostname if not set
+            if not process.system("hostname localhost.localdomain", sudo=True):
+                self.error("Failed to set hostname\n")
+
+        self.hostname = process.run("hostname", verbose=False,
+                                    env={"LANG": "C"}).stdout
+        # Backup info required to restore hostname
+        self.workdir = data_dir.get_tmp_dir()
+        self.backup_dir = os.path.join(self.workdir, 'net-tools')
+        if not os.path.isdir(self.backup_dir):
+            os.makedirs(self.backup_dir)
+
+        self.hostname_file = os.path.join(self.backup_dir, "HOSTNAME")
+        self.myhostname_file = os.path.join(self.backup_dir, "MYHOSTNAME")
+        fobj = open(self.hostname_file, 'w')
+        fobj.write(self.hostname)
+        fobj.close()
+        fobj = open(self.myhostname_file, 'w')
+        fobj.write("myhost.my-domain")
+        fobj.close()
+
+    def test_hostname(self):
+        """
+        i.Verifies hostname command options and
+        ii.Test to change the hostname
+        """
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            self.error("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            output = process.run("hostname -%s" % option, verbose=False)
+            if not output.stdout:
+                self.error("unexpected response from hostname command")
+            self.log.info("verified -%s option of hostname command" % option)
+
+        # Test to change hostname
+        process.system("hostname -F %s" % self.myhostname_file,
+                       verbose=False,
+                       sudo=True)
+        self.restore_hostname = True
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if "myhost.my-domain" not in output.stdout:
+            self.error("unexpected response from hostname -F command and " +
+                       "hostname -F didn't set hostname")
+        ret = self._restore_hostname(self.hostname_file, self.hostname)
+        if ret:
+            self.error(ret)
+        self.restore_hostname = False
+
+    @staticmethod
+    def _restore_hostname(hostnamefile, hostname):
```

Comment:
Well if you want this function only to work as `restore_hostname` then I think it should not accept the arguments and it should use `self` to get the correct values. On the other hand you could also call this `set_hostname_file` and it can be used even for setting the hostname :wink:.